### PR TITLE
Sitka factory implementation

### PR DESCRIPTION
**Pull request by jIrwinCline**
implement sitka factory in sitka-test.ts

Reorganize module and store tests to work with sitka factory

### CODE REVIEW

File: tests/__tests__/sitka_utils.test.ts [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 import { createAppStore } from "../../src/sitka"
 import rewire from "rewire"
 
 const utils = rewire("../../dist/sitka.js")
 
-const { text: textModule } = sitka.getModules()
-
 describe("Sitka Util Functions", () => {
   test(`createAppStore returns Redux store`, () => {
+    // todo: using sitka factory results in referenceError: alert is not defined
```
Comment (jIrwinCline) [inline tests/__tests__/sitka_utils.test.ts:9]: related to issue testing for - running test passes but results in referenceError: alert not defined?

this happens when switching this test to use sitkaFactory

File: tests/__tests__/sitka_utils.test.ts [hunk 2]
```diff
@@ -1,13 +1,12 @@
...
 import { createAppStore } from "../../src/sitka"
 import rewire from "rewire"
 
 const utils = rewire("../../dist/sitka.js")
 
-const { text: textModule } = sitka.getModules()
-
 describe("Sitka Util Functions", () => {
   test(`createAppStore returns Redux store`, () => {
+    // todo: using sitka factory results in referenceError: alert is not defined.  Test will still pass
```
Comment (jIrwinCline) [inline tests/__tests__/sitka_utils.test.ts:9]: Error presents itself when switching over to use sitkaFactory here. referenceError: alert not defined.

related to same error when we switch provideForks test to sitkaFactory (on diffferent branch)

File: package.json [hunk 3]
```diff
@@ -12,6 +12,7 @@
   },
   "scripts": {
     "test": "yarn build && jest",
+    "detectOpenHandles": "yarn build && jest --detectOpenHandles",
```
Comment (rachoac) [inline package.json:15]: probably don't need to check this particular test command in

File: tests/__tests__/sitka_module.test.ts [hunk 4]
```diff
@@ -1,98 +1,105 @@
...
+    const actual = sitka.getModules().text.getStateTestDelegate(allState)
+    expect(actual).toEqual(defaultTextModuleState)
   })
 
   test('mergeState sets partial state of module', () => {
     // handleUpdateSize implements merge state
-    textModule.handleUpdateSize(5)
-    const { text: actual } = store.getState()
+    const { store, sitka } = createSitkaAndStore()
+    sitka.getModules().text.handleUpdateSize(5)
```
Comment (rachoac) [inline tests/__tests__/sitka_module.test.ts:27]: i don't understand why updating size to 5 will make expected numberOfEdits to 0?

File: package.json (hunk 3 above)
Comment (jIrwinCline) [reply package.json:15]: Dustin added this to combat a warning we were getting. do you want it removed definitely?

File: tests/__tests__/sitka_module.test.ts [hunk 5]
```diff
@@ -102,29 +109,36 @@ describe("SitkaModule", () => {
 
  describe('provideMiddleware adds middleware to Sitka', () => {
 
-   test('middleware provided from a sitka module adds middleware to Sitka', () => {
-      const actual = sitka.createSitkaMeta().middleware[0]
-      expect(actual).toEqual(textModule.historyMiddleware)
-   })
+  test('middleware provided from a sitka module adds middleware to Sitka', () => {
```
Comment (rachoac) [inline tests/__tests__/sitka_module.test.ts:112]: probably need a test to confirm that no middleware is added to sitka if log is not specified, and history is not specified

File: tests/__tests__/sitka_module.test.ts (hunk 4 above)
Comment (jIrwinCline) [reply tests/__tests__/sitka_module.test.ts:27]: tightened scope

File: tests/__tests__/sitka_utils.test.ts (hunk 2 above)
Comment (jIrwinCline) [reply tests/__tests__/sitka_utils.test.ts:9]: fixed

File: package.json (hunk 3 above)
Comment (rachoac) [reply package.json:15]:
```
A worker process has failed to exit gracefully and has been force exited. This is likely caused by tests leaking due to improper teardown. Try running with --detectOpenHandles to find leaks.
```
that is a bug we need to file in the backlog
4:34
make a note that yarn build && jest --detectOpenHandles is involved
4:34
and then remove that from package.json

File: tests/__tests__/sitka_module.test.ts (hunk 5 above)
Comment (jIrwinCline) [reply tests/__tests__/sitka_module.test.ts:107]: added

**Pull request merged**